src/lib.rs

```diff
@@ -509,6 +511,107 @@ impl<'a, T: CBufItem, const SIZE: usize> CBufReader<'a, T, SIZE> {
         self.next_local
     }
 
+    /// Sets the index used for sequential reads.
+    ///
+    /// (i.e., [`fetch_next_item`](Self::fetch_next_item)
+    /// and [`available_items_iter`](Self::available_items_iter)).
+    #[inline]
+    pub fn set_next_index(&mut self, next: CBufIndex<SIZE>) {
```

Comment:
I'm not sure whether we should have this method or not. See my comment at [line 532] for more details.

[line 532]: https://github.com/BlackCAT-CubeSat/bc-buf/pull/3/commits/fb6cd628e6749ce98967774d339cd30158d9d29f#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R532

Reply:
Would it be better to replace it with:
```
pub fn reader_starting_at(&self, start: CBufIndex<SIZE>) -> Self;
```

With a lifetime of the returned `CBufReader` that depends on that of the underlying `CBuf`, and not on the `self`.

Reply:
Yes, and as we discussed yesterday, `Self` is the appropriate return type here.

Reply:
Look at my implementation of `reader_starting_at`.  Is the `self.copy()` the right thing?

Reply:
Per the [standard library docs](https://doc.rust-lang.org/std/clone/trait.Clone.html#derivable), the `Clone` impl `#[derive(Clone)]` generates does a field-wise `Clone`, which is OK in this case.

The bigger issue is that the traits a type implements are part of its public API/ABI. As such, my question is: should `CBufReader` be `Clone` for the crate's users? Answering my own question, given what it is and what it does, I think that's OK, so you're good adding `#[derive(Clone)]` on `CBufReader`.

Incidentally, if you didn't want to have `CBufReader` implement `Clone`, something like the following [would work](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax):

```rust
let result = CBufReader {
    next_local: next,
    ..self
};
```